### PR TITLE
Add 'scan_type' to the RunStart document.

### PR DESCRIPTION
**Pull request by ericdill**
Same as NSLS-II/metadatastore#190

### CODE REVIEW

**Comment by tacaswell** [comment]
closed/opened to re-start travis based on master now, not when it was created

```python
======================================================================
ERROR: databroker.tests.test_misc.test_pims_images_old_api
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/mc/envs/testenv/lib/python3.5/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/home/travis/build/NSLS-II/databroker/databroker/tests/test_misc.py", line 21, in test_pims_images_old_api
    assert_equal(len(images), image_and_scalar.num1)
AttributeError: module 'databroker.examples.sample_data.image_and_scalar' has no attribute 'num1'
======================================================================
ERROR: databroker.tests.test_misc.test_pims_images
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/mc/envs/testenv/lib/python3.5/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/home/travis/build/NSLS-II/databroker/databroker/tests/test_misc.py", line 30, in test_pims_images
    assert_equal(len(images), image_and_scalar.num1)
AttributeError: module 'databroker.examples.sample_data.image_and_scalar' has no attribute 'num1'
```

File: databroker/examples/sample_data/image_and_scalar.py [hunk 1]
```diff
@@ -13,9 +13,9 @@
 
 
 # number of motor positions to fake
-num1 = 20
```
Comment (tacaswell) [inline databroker/examples/sample_data/image_and_scalar.py]: this is referenced in the tests.  Either update the tests or put an alias back in.

**Comment by danielballan** [comment]
There is a rename coming on scan_type. Moving to the word plan, with input from @klauer -- and the idea that each run involves scanning something is not true across the board.

**Comment by danielballan** [comment]
@ericdill If I were you I would just go with using bluesky to write the test data, but if you'd rather see this merged change `scan_type` -> `plan_type`.

**Comment by ericdill** [comment]
@danielballan I like the idea of using bluesky to write test data.  Any objections to moving (and simplifying) the example code for generating data into bluesky from metadatastore, bluesky and filestore? I'm all for separation of concerns, but it seems like the example generation code is just always lagging behind

**Comment by danielballan** [comment]
Off the top of my head, I think that sounds fine. To be clear, that would make bluesky a test-only dependency of these projects. We have been trying the alternative -- imperfectly imitating bluesky's output by hand -- for awhile now and I think it's clear that it is worse. But I would want a group consensus on that....

cc @klauer @tacaswell @dchabot

**Comment by klauer** [comment]
Seems to make more sense generating it with bluesky. But what package dependency implications would that have?

**Comment by danielballan** [comment]
@ericdill I think you are using the word "test dependency" to mean "test dependency listed in the conda package." I was using it to mean "package you need to install if you want to run the tests." So I want to make sure that everyone is OK that bluesky would become *de facto* required for any developer working on metadatastore.

**Comment by ericdill** [comment]
> @ericdill I think you are using the word "test dependency" to mean "test dependency listed in the conda package." 

Yes, I was attempting to answer the question of 'what would `conda install metadatastore` do', but I guess I'm not entirely sure that *was* the question that @klauer was even asking. Oh well.

> I was using it to mean "package you need to install if you want to run the tests." So I want to make sure that everyone is OK that bluesky would become de facto required for any developer working on metadatastore.

Yes, I understand.  To further hammer on this idea, since the bluesky RunEngine is currently the only way to get experimental data into the databroker stack, I am like one million :+1: on making bluesky the only way to get test data into the testing databroker stack so we don't have to keep twiddling the data generation code in mds/db.

**Comment by danielballan** [comment]
Ah, I missed @klauer's question. Developers working on metadatastore/filestore/databroker would have to install bluesky and its dependencies -- which, thanks to the callbacks, are pretty sprawling right now. But they would not need bluesky's *test* dependencies (like ophyd).

**Comment by ericdill** [comment]
closing old prs

**Pull request closed without merging**